### PR TITLE
Subnav (v2): Improve styling of child links

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Applies all changes from https://github.com/strapi/design-system/pull/635 to the v2 version of the component.

### Why is it needed?

To be in sync between v1 and v2.

### CODE REVIEW

File: packages/strapi-design-system/src/SubNav/SubNavLink.js [hunk 1]
```diff
@@ -88,5 +88,6 @@ SubNavLink.propTypes = {
   icon: PropTypes.element,
   isSubSectionChild: PropTypes.bool,
   link: PropTypes.element,
+  subSectionChild: PropTypes.bool,
```
Comment (ronronscelestes) [inline packages/strapi-design-system/src/SubNav/SubNavLink.js:91]: should it be `isSubSectionChild`?

**Review by ronronscelestes — CHANGES_REQUESTED** [review]
Found a small typo. Other than that looks good, thank you!

**Review by HichamELBSI — APPROVED** [review]
LGTM

File: packages/strapi-design-system/src/SubNav/SubNavLink.js (hunk 1 above)
Comment (gu-stav) [reply packages/strapi-design-system/src/SubNav/SubNavLink.js]: Good catch. I messed up the rebease - should be fixed.

**Review by ronronscelestes — APPROVED** [review]
LGTM 🌞💅

**Pull request merged**